test/GitIgnoreTest.php

```diff
@@ -91,13 +119,22 @@ public function testAddEntryDoesNotAddFileOrDirectoryIfItIsInsideAnIgnoredDirect
 
     public function testIgnoreDirectoriesAreNotWrittenIfNoEntriesAreAdded()
```

Comment:
With that test too, it appears that what you have there is still valid. In that test, there were no entries at the start. You provided no further entries, so the file was not written.

Correct me if I'm wrong, but what was happening before was that no entries were written **even if** you added new ignored directories. That didn't make this test incorrect, but it didn't capture the edge case that existed. Even with the new changes, this test is still valid.